app/uk/gov/hmrc/cdsreimbursementclaimfrontend/controllers/claims/CheckClaimantDetailsController.scala

```diff
@@ -41,30 +56,97 @@ class CheckClaimantDetailsController @Inject() (
+        val emptyForm  = checkClaimantDetailsAnswerForm
+        val filledForm = answers.fold(emptyForm)(emptyForm.fill(_))
+        Ok(renderTemplate(filledForm, fillingOutClaim, router))
       }
     }
 
+  def submit(implicit journey: JourneyBindable): Action[AnyContent] =
+    authenticatedActionWithSessionData.async { implicit request =>
+      withAnswersAndRoutes[CheckClaimantDetailsAnswer] { (fillingOutClaim, _, router) =>
+        checkClaimantDetailsAnswerForm
+          .bindFromRequest()
+          .fold(
+            formWithErrors => BadRequest(renderTemplate(formWithErrors, fillingOutClaim, router)),
+            formOk => {
+              val newDraftClaim  = fillingOutClaim.draftClaim.fold(_.copy(checkClaimantDetailsAnswer = Option(formOk)))
+              val updatedJourney = fillingOutClaim.copy(draftClaim = newDraftClaim)
+
+              EitherT(updateSession(sessionStore, request)(_.copy(journeyStatus = Some(updatedJourney))))
+                .leftMap(err => Error(s"Could not save Declarant Type: ${err.message}"))
+                .fold(
+                  e => {
+                    logger.warn("Submit Declarant Type error: ", e)
+                    errorHandler.errorResult()
+                  },
+                  _ => Redirect(router.nextPageForClaimantDetails())
+                )
+            }
+          )
+      }
+
+    }
+
+  def renderTemplate(
+    form: Form[CheckClaimantDetailsAnswer],
+    fillingOutClaim: FillingOutClaim,
+    router: ReimbursementRoutes
+  )(implicit
+    request: RequestWithSessionData[_],
+    messages: Messages,
+    viewConfig: ViewConfig
+  ): HtmlFormat.Appendable =
+    claimantDetails(
+      form,
+      extractContactsRegisteredWithCDSA(fillingOutClaim),
+      extractEstablishmentAddress(fillingOutClaim),
+      extractContactDetails(fillingOutClaim),
+      extractContactAddress(fillingOutClaim),
+      router
+    )
+
 }
 
 object CheckClaimantDetailsController {
   val languageKey: String = "claimant-details"
 
+  sealed trait CheckClaimantDetailsAnswer extends Product with Serializable
+
+  case object UseContactDetails extends CheckClaimantDetailsAnswer
+  case object UseDetailsRegisteredWithCDS extends CheckClaimantDetailsAnswer
+
+  val checkClaimantDetailsAnswerForm: Form[CheckClaimantDetailsAnswer] =
+    Form(
+      mapping(
+        languageKey -> number
+          .verifying("invalid", a => a === 0 || a === 1)
+          .transform[CheckClaimantDetailsAnswer](
+            value =>
+              if (value === 0) UseContactDetails
+              else UseDetailsRegisteredWithCDS,
+            {
+              case UseContactDetails           => 0
+              case UseDetailsRegisteredWithCDS => 1
+            }
+          )
+      )(identity)(Some(_))
+    )
+
+  implicit val format: OFormat[CheckClaimantDetailsAnswer] = derived.oformat[CheckClaimantDetailsAnswer]()
+
   def extractContactsRegisteredWithCDSA(fillingOutClaim: FillingOutClaim): NamePhoneEmail = {
```

Comment:
A better name would be extractDetailsRegisteredWithCDS 
Is the 'A' on the end of 'extractContactsRegisteredWithCDSA' a typo?